testing/src/scenario/_runtime.py

```diff
@@ -350,6 +350,9 @@ def exec(
                 # If ops raised _Abort(0) then we want to treat that as normal completion.
                 if e.exit_code != 0:
                     raise
+                # If _Abort was raised before we created the instance, we can't get the state, so
+                # still want to fail.
+                assert ops is not None
```

Comment:
Hmm, so we can get `_Abort` even during the `Ops` call? (before creating the instance)

Raising an assertion error feels wrong in that case. I'd rather see:
```suggestion
                if ops is None:
                    raise
```

On the other hand, if we can't actually get `_Abort` from `Ops`, then I'd prefer to see the `try ... except` restructured here. For example, a nested `try`, `except _Abort` and `finally` for the cleanup conditional on `ops` existing could be wrapped around just `yield ops`.